models/article.model.js

```
@@ -48,19 +48,55 @@ exports.updateArticle = (articleId,newVote) => {
 }
 
 
-exports.fetchAllArticles = () => {
+exports.fetchAllArticles = (sort_by="created_at",order="desc",topic) => {
 
-    return db.query(`
+    let baseQueryStr = `
     SELECT articles.* ,
     COUNT(comments.comment_id) ::INT AS comment_count
     FROM articles 
     LEFT JOIN comments
     ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY created_at DESC;
-    `).then((articles)=> {
-        return articles.rows
-    })
+    `
+
+    const validSortBy = ["article_id","title", "topic", "author","body","created_at","vote"];
+
+    let sortByStr = 'ORDER BY '
+    if (validSortBy.includes(sort_by)) {
+    sortByStr += sort_by + " "
+    }
+    else {
+    sortByStr += "created_at "
+    }
+
+    let orderStr = ""
+    if (order === "asc" || order === "ASC") {
```

What if order was 'dljsjdasdsdffsdf' - this logic would mean it becomes 'desc'. - but would need to throw an error